packages/driver/src/cy/commands/files.ts

```diff
@@ -4,118 +4,132 @@ import { basename } from 'path'
 import $errUtils from '../../cypress/error_utils'
 import type { Log } from '../../cypress/log'
 
-interface InternalReadFileOptions extends Partial<Cypress.Loggable & Cypress.Timeoutable> {
-  _log?: Log
-  encoding: Cypress.Encodings
+interface ReadFileOptions extends Partial<Cypress.Loggable & Cypress.Timeoutable> {
+  encoding?: Cypress.Encodings
 }
 
 interface InternalWriteFileOptions extends Partial<Cypress.WriteFileOptions & Cypress.Timeoutable> {
   _log?: Log
 }
 
 export default (Commands, Cypress, cy, state) => {
-  Commands.addAll({
-    readFile (file, encoding, userOptions: Partial<Cypress.Loggable & Cypress.Timeoutable> = {}) {
-      if (_.isObject(encoding)) {
-        userOptions = encoding
-        encoding = undefined
+  Commands.addQuery('readFile', function readFile (file, encoding, options: ReadFileOptions = {}) {
```

Comment:
I would expect this work to be done in its own pull request to make this review smaller. I wouldn't expect this change to impact should() & and().

Reply:
I'll move this to its own PR, good call.

It ends up being a prerequisite for updating .should() / .and(). Part of `.readFile()`'s behavior is that if the following assertions fail, it needs to re-read the file from disk - which is basically the idea behind queries! With how assertions work changing, .readFile() needs to fall into the new system as well.

I'll put more details in the description of the new PR.

Reply:
https://github.com/cypress-io/cypress/pull/25595 contains these changes stand-alone.